libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.spec.ts

```diff
@@ -91,6 +91,50 @@ describe('ModalWrapperComponent', () => {
     });
   });
 
+  describe('with interact with background', () => {
+    beforeEach(async(async () => {
+      TestHelper.resizeTestWindow(TestHelper.screensize.desktop);
+      spectator = modalWrapperTestBuilder
+        .flavor('drawer')
+        .interactWithBackground()
+        .build();
+    }));
+
+    it('should adapt modal size to children after ion-modal has been presented', async () => {
+      spectator.element.style.height = '500px';
+      spectator.element.style.width = '400px';
+
+      spectator.component.config.interactWithBackground = true;
+
+      spectator.component['ionModalDidPresent'].next();
+      spectator.component['ionModalDidPresent'].complete();
+
+      await TestHelper.waitForTimeout();
+      spectator.detectChanges();
+
+      expect(spectator.component['ionModalElement'].style.top).toBe('330px');
+      expect(spectator.component['ionModalElement'].style.left).toBe('400px');
+      expect(spectator.component['ionModalElement'].style.right).toBe('400px');
+    });
+
+    it('should adapt modal size to children on resize', async () => {
+      spectator.element.style.height = '500px';
+      spectator.element.style.width = '400px';
+
+      spectator.component.config.interactWithBackground = true;
+
+      spectator.component['viewportResize'].next();
+      spectator.component['viewportResize'].complete();
+
+      await TestHelper.waitForResizeObserver();
+      spectator.detectChanges();
+
+      expect(spectator.component['ionModalElement'].style.top).toBe('330px');
+      expect(spectator.component['ionModalElement'].style.left).toBe('400px');
+      expect(spectator.component['ionModalElement'].style.right).toBe('400px');
```

Comment:
Are these related to `TestHelper.screensize.desktop`? If so, consider readability and calculate/somehow convey the relation/expected values...